xray/services/scan.go

```diff
@@ -162,6 +195,59 @@ func (ss *ScanService) GetScanGraphResults(scanId string, includeVulnerabilities
 	return &scanResponse, err
 }
 
+func (ss *ScanService) xrayToXscUrl() string {
+	return strings.Replace(ss.XrayDetails.GetUrl(), "/xray", "/xsc", 1)
+}
+
+func (ss *ScanService) SendScanGitInfoContext(details *XscGitInfoContext) (multiScanId string, err error) {
+	httpClientsDetails := ss.XrayDetails.CreateHttpClientDetails()
+	utils.SetContentType("application/json", &httpClientsDetails.Headers)
+	requestBody, err := json.Marshal(details)
+	if err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	url := ss.xrayToXscUrl() + postScanContextAPI
+	resp, body, err := ss.client.SendPost(url, requestBody, &httpClientsDetails)
+	if err != nil {
+		return
+	}
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusCreated); err != nil {
+		return
+	}
+	scanResponse := XscPostContextResponse{}
+	if err = json.Unmarshal(body, &scanResponse); err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	return scanResponse.MultiScanId, err
+}
+
+// IsXscEnabled will try to get XSC version. If route is not available, user is not entitled for XSC.
+func (ss *ScanService) IsXscEnabled() (xsxVersion string, err error) {
+	httpClientsDetails := ss.XrayDetails.CreateHttpClientDetails()
+	serverDetails := ss.XrayDetails
+	xscUrl := strings.Replace(serverDetails.GetUrl(), "xray", "xsc", 1)
+	resp, body, _, err := ss.client.SendGet(xscUrl+XscVersionAPI, true, &httpClientsDetails)
+	if err != nil {
+		err = errorutils.CheckErrorf("failed to get XSC version, response: " + err.Error())
+		return
+	}
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusOK, http.StatusNotFound); err != nil {
```

Comment:
StatusNotFound ?

Reply:
When you don't have XSC it will resolve in StatusNotFound, which is expected in most cases.